autoload/LanguageClient.vim

```diff
@@ -1289,6 +1299,13 @@ function! LanguageClient#serverStatusMessage() abort
     return g:LanguageClient_serverStatusMessage
 endfunction
 
+" Set to 1 when the language server is running for current buffer
+let g:LanguageClient_isServerRunning = 0
```

Comment:
This is a global variable while we use it as a buffer local one.

I'd suggest we move it to buffer local. Thus we can update it accordingly with various lifecycle autocmds such as BufType, BufNewFile etc.

Reply:
I'll change it to buffer local variable and init `b:LanguageClient_isServerRunning` to 0 in `handleBufEnter`, is it ok?